src/Documentation/2.0/Transactions.md

```diff
@@ -1,6 +1,92 @@
----
+__---__
 layout: page
 title: Transactions in Orleans 2.0
 ---
 
-# Placeholder for Transactions documentation
\ No newline at end of file
+# Orleans Transactions (Beta)
+__Orleans transactions are in a beta state and should not be used in production systems.__
+Orleans supports distributed transactions against persistent grain state.
+## Setup
+Orleans transactions are opt-in.  A silo must be configured to use transactions.  If it is not, any calls to transactional grains will receive an OrleansTransactionsDisabledException.
+### Transaction Manager
+Orleans transactions requires a transaction manager which is authoritative on the state of a transaction.  Currently, the transaction manager is run within the cluster and must be configured.  This can be programmatically configured on the silo host builder using UseInClusterTransactionManager and TransactionsConfiguration.
+The TransactionsConfiguration contains the following configurable values:
+* TransactionIdAllocationBatchSize - To avoid a storage call on every transaction start, transaction Ids are allocated in batches. This is the number of new transaction Ids allocated per transaction Id generation request (storage call).
+* AvailableTransactionIdThreshold - A new batch of transaction Ids will be automatically allocated if the available ids drop below this threshold.
+* TransactionRecordPreservationDuration - How long to preserve a transaction record in the TM memory after the transaction has completed.  This is used to answer queries about the outcome of the transaction.
+It is suggested that you use the defaults for these values unless they demonstrate themselves as insufficient.
+### Transaction Log
+The transaction manager must persist transaction results for recoverability purposes.  To support a range of backend storage solutions an abstraction has been provided for log storage.  We currently support an azure storage version of the log storage, and an in-memory version for development and test purposes.  This can be programmatically configured on the silo host builder using UseAzureTransactionLog or UseInMemoryTransactionLog.
+### Transactional State
+To support various transactional storage patterns the grain facing interface and supporting logic is pluggable.  We currently only support ITransactionalState.  This can be programmatically enabled on the silo host builder using UseTransactionalState.  To use a transactional state, a storage provider needs also be configured.
+Example:
+       var config = ClusterConfiguration.LocalhostPrimarySilo();
+       config.AddAzureTableStorageProvider(=”<storage_account>”);
+
+       var builder = new SiloHostBuilder()
+       .UseConfiguration(config)
+       .AddApplicationPartsFromReferences(typeof(ATMGrain).Assembly)
```

Comment:
Unfortunately, this line is out-dated already. It should be `.ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(ATMGrain).Assembly).WithReferences())`